app/controllers/comments_controller.rb

```diff
@@ -230,6 +230,20 @@ def unhide
     end
   end
 
+  def admin_delete
+    @comment = Comment.find(params[:comment_id])
+    authorize @comment
+    @comment.deleted = true
+
+    if @comment.save
+      redirect = @comment.commentable&.path
+      redirect_to URI.parse(redirect).path, notice: "Comment was successfully deleted."
```

Comment:
`URI.parse` here will throw an error if redirect is nil. We might want an index fallback path here if that is the case. 
```
[11] pry(main)> URI.parse(nil)
URI::InvalidURIError: bad URI(is not URI?): nil
from /Users/mollystruve/.rvm/rubies/ruby-2.7.1/lib/ruby/2.7.0/uri/rfc3986_parser.rb:18:in `rescue in split'
Caused by NoMethodError: undefined method `to_str' for nil:NilClass
from /Users/mollystruve/.rvm/rubies/ruby-2.7.1/lib/ruby/2.7.0/uri/rfc3986_parser.rb:16:in `split'
```

Reply:
This code path won't execute if `redirect` is `nil` because of `@comment.save && redirect`. That said, I don't think we need to parse at all:

```ruby
URI.parse(redirect).path == redirect
#=> true
```

`redirect` is already a path (due to `&.path`), so parsing it and then calling `.path` on it will result in the same value.